gradle-baseline-java/src/main/java/com/palantir/baseline/plugins/BaselineModuleJvmArgs.java

```diff
@@ -111,19 +114,14 @@ private void applyToJavaProject(Project project) {
         });
 
         project.getTasks().withType(Test.class).configureEach(test -> {
-            ModuleJvmArgsArgumentProvider provider = newModuleJvmArgsArgumentProvider(
-                    project, extension, project.files((Callable<FileCollection>) test::getClasspath), test.getName());
-            test.getJvmArgumentProviders().add(provider);
+            test.getJvmArgumentProviders()
+                    .add(ModuleJvmArgsArgumentProvider.fromClasspathAndExtension(test, test::getClasspath));
             setTaskInputsFromExtension(test, extension);
```

Comment:
from IRL conversation, I don't think these are necessary anymore. I imagine not for task dependencies, but caching perhaps? since prior to  #3119 we weren't using the argument provider which with a managed type which is `@Nested`. Probably not required for this PR, but can be done independently of it, @FinlayRJW for SA.